### PR TITLE
[ISV-3498] Create an end-to-end test for the community operator pipeline

### PR DESCRIPTION
**Pull request by ezopezo**
E2e testing logic implementation.

Change-log:
- [operator test data] Community operator data for testing: https://github.com/redhat-openshift-ecosystem/operator-pipelines-test/tree/e2e-community-test-operator/operators/test-e2e-community-operator.
- Custom base branch (main for certified operators logic) bearing config.yaml with pointer to community-operators index has been created and referenced in community-operators-integration-tests playbook; https://github.com/redhat-openshift-ecosystem/operator-pipelines-test/blob/community/config.yaml
- Prow tasks were disconnected from community hosted pipeline for the sake of of current design changes. Complete clean-up will be done separately in [ISV-4143].
- Community event listener interceptor [ISV-4140] and two certified operator tasks (submission-validation and verify-changed-directories) [ISV-4196] requests were parameterized with github authentication token to avoid github API rate limiting. This had to be done within this PR enabling effective development.

Deprecated:
- [CLOSED] Prow job configuration: https://github.com/openshift/release/pull/44011 - closed because of current design changes (using preflight instead own prow solution). 
- [REMOVED] Temp base branch for prow PR test: https://github.com/redhat-openshift-ecosystem/community-operators-pipeline-preprod/compare/integration-tests-community-operators - removed, all testing data and testing-PR-execution-env delegated to operator-pipelines-test repository.
- Whole integration tests logic is now running on PR but also after merging PR as previously. Community tests and certified operator tests are running in parallel, but within different branches and ocp namespaces. Delegated to ISV-4199 due to bloated scope of this task and security concerns.

### CODE REVIEW

File: ansible/inventory/group_vars/operator-pipeline-integration-tests.yml
Comment (Allda) [inline ansible/inventory/group_vars/operator-pipeline-integration-tests.yml:27]: Have you verified if this is really working? The webhook for isv tests has a predefined URL based on a pipeline name. The same webhook for the community has a different URL.

File: ansible/playbooks/community-operators-integration-tests.yaml
Comment (Allda) [inline ansible/playbooks/community-operators-integration-tests.yaml:56]: Nginx is not needed for the community. You can drop it.

File: .github/workflows/common/prepare-and-install-deps.yml
Comment (mporrato) [inline .github/workflows/common/prepare-and-install-deps.yml:25]: Please properly quote strings containing variables or command substitutions where possible.
(See https://www.shellcheck.net/wiki/SC2086)

File: .github/workflows/integration-tests.yml
Comment (mporrato) [inline .github/workflows/integration-tests.yml:91]: Please add quotes here as well.

File: ansible/roles/integration-tests/tasks/community_pipeline.yaml
Comment (Allda) [inline ansible/roles/integration-tests/tasks/community_pipeline.yaml:6]: I think this can be dropped since the PR opening triggers the pipeline.

File: ansible/roles/integration-tests/tasks/open_pull_request.yaml
Comment (Allda) [inline ansible/roles/integration-tests/tasks/open_pull_request.yaml:5]: Doesn't this line leak token when using a verbose mode?

Comment (ezopezo) [reply ansible/roles/integration-tests/tasks/open_pull_request.yaml:5]: Good point, I add no_log statement

File: ansible/roles/integration-tests/tasks/community_pipeline.yaml
Comment (ezopezo) [reply ansible/roles/integration-tests/tasks/community_pipeline.yaml:6]: Yes you're right, I planned to do it as final clean up

File: .github/workflows/integration-tests.yml
Comment (Allda) [inline .github/workflows/integration-tests.yml:77]: Is this secret needed for the community pipeline if it doesn't have a CI pipeline?

File: .github/workflows/integration-tests.yml
Comment (Allda) [inline .github/workflows/integration-tests.yml:83]: Same here.

Comment (ezopezo) [reply .github/workflows/integration-tests.yml:83]: Yep, you are right, good catch, removing.

File: operator-pipeline-images/operatorcert/github.py
Comment (Allda) [inline operator-pipeline-images/operatorcert/github.py:68]: Is there any reason for removing the debug log?

**Review by Allda — APPROVED** [review]
Few minor comments, but otherwise it looks okay. Please wait for a second approval since this PR is not tiny.

**Comment by Allda** [comment]
@ezopezo Please don't forget to squash before a merge so we don't end up with test commits in the main. You can select a merge strategy when merging the PR.

File: operator-pipeline-images/operatorcert/github.py
Comment (ezopezo) [reply operator-pipeline-images/operatorcert/github.py:68]: Nope, misunderstanding, thanks for reminding

File: ansible/roles/integration-tests/tasks/open_pull_request.yaml
Comment (mporrato) [inline ansible/roles/integration-tests/tasks/open_pull_request.yaml:21]:
```suggestion
- name: Create a pull request
  uri:
    url: "https://api.github.com/repos/{{ git_upstream_repo }}/pulls"
    method: POST
    headers:
      Authorization: "token {{ ci_pipeline_github_personal_access_token }}"
      Accept: application/vnd.github.v3+json
    body_format: json
    body:
      title: "operator {{ operator_package_name }} ({{ operator_bundle_version }})"
      body: E2e test for community operators
      head: "{{ git_bundle_branch }}"
      base: "{{ git_upstream_branch }}"
  register: pr_response
  no_log: true

- name: Display PR response
  debug:
    var: pr_response
```

File: ansible/roles/integration-tests/tasks/prepare_ci_file.yaml
Comment (mporrato) [inline ansible/roles/integration-tests/tasks/prepare_ci_file.yaml:22]:
```suggestion
    git checkout "$UPSTREAM_BRANCH"

    # Fetch the ci.yaml from SRC_BRANCH and place it in the working directory
    git checkout "origin/$SRC_BRANCH" -- "operators/$OPERATOR_PACKAGE_NAME/ci.yaml"

    git add "operators/$OPERATOR_PACKAGE_NAME/ci.yaml"
    git commit -m "Copied ci.yaml from $SRC_BRANCH"
    git push origin "$UPSTREAM_BRANCH"
```

**Review by mporrato — APPROVED** [review]

File: .github/workflows/integration-tests.yml
Comment (ezopezo) [reply .github/workflows/integration-tests.yml:112]: Returned back, due to findings summarized in comment.

File: .github/workflows/integration-tests.yml
Comment (ezopezo) [reply .github/workflows/integration-tests.yml:120]: Returned back, due to findings summarized in comment.

**Pull request merged**
